### ISSUE TITLE
Godot 4.1rc2 Weird floating plane mesh and black flickering screen.

### ISSUE THREAD

**Issue by NancokPS2** [godotengine/godot#78936]
### Godot version

Godot 4.1rc2

### System information

Windows 10 x64 - Godot v4.1rc2 - OpenGL (Compatibility) - dedicated. GPU: FirePro v4900. CPU: AMD Phenom II X6 

### Issue description

![Untitled](https://github.com/godotengine/godot/assets/55665720/4238db38-1dcd-4f2e-8782-62e9c5c3e3b7)
The black flickering screen happens when i open any project. But the floating rectangle only does when using shaders/materials.  
Setting a WorldEnvironment seems to briefly fix it but it returns shortle after.  
  
At first i tought the rectangle moved at random, but it seems to only happen a short time after i move the camera. It snaps to a seemingly random position and deforms when it does tho.

### Steps to reproduce

- Open project, see black flickering screen and other similar artifacts.    
  
------------------------------------

- Create a MeshInstance3D  
- Add a Material, add a ShaderMaterial as a next_pass

### Minimal reproduction project

N/A

**Comment by NancokPS2** [comment]
In case anyone wonders. There's no AutoLoads and the shader code shown is what there is, the ending bracket got cut in the screenshot but that's about it.

**Comment by Calinou** [comment]
The [FirePro V4900](https://www.techpowerup.com/gpu-specs/firepro-v4900.c578) is a very old GPU with end-of-life drivers, so I wouldn't be surprised if OpenGL 3.3 can't work well on it. See also https://github.com/godotengine/godot/issues/78888.

**Comment by NancokPS2** [comment]
> 

Yeah, i know it's old. Altho it can run games that require OpenGL 4.5 just fine so i can confirm that it has support for that.

**Comment by clayjohn** [comment]
Could you please upload an MRP so that other contributors can try to reproduce the issue? I haven't seen an issue like this before, but there could easily be something in your shader that is causing this behaviour

**Comment by NancokPS2** [comment]
> Could you please upload an MRP so that other contributors can try to reproduce the issue? I haven't seen an issue like this before, but there could easily be something in your shader that is causing this behaviour

Sorry for the ignorance, but what is an MRP?  
  
Also, the display still flickers without shaders.

**Comment by Calinou** [comment]
> Sorry for the ignorance, but what is an MRP?

A MRP (minimal reproduction project) is a project that reproduces the issue. The project must run on its own but should not have unnecessary files included.

**Comment by NancokPS2** [comment]
I kind left this to rot here. But as far as MRP, this doesn't require anything in particular. Just adding a mesh and adding a new material of ANY kind will cause the bug.   

I updated the bug to mention the latest version, because it keeps happening as of the latest versions, both stable and experimental.  
  
It would seem as if Godot is unable to compile shaders during runtime, but it can do so when booting up. Any shaders compiled while loading the project will work, any attempt at recompiling them later fails.
